napari/utils/event_handler.py

```diff
@@ -40,15 +43,18 @@ def on_change(self, event=None):
         # until refactor on all layers is complete, not all events will have a
         # value property
         try:
+            # print(event.type, event.value, self.components)
             value = event.value
             logger.debug(f" value: {value}")
         except AttributeError:
             logger.debug(f" did not handle event {event.type}")
             return
 
         # Update based on event value
-        for component in self.components_to_update:
-            update_method_name = f"_on_{event.type}_change"
-            update_method = getattr(component, update_method_name, None)
-            if update_method:
-                update_method(value)
+        for componentref in self.components:
+            component = componentref()
```

Comment:
As above — add a comment here reminding readers that `self.components` contains weakrefs rather than refs.

I agree with Talley that this doesn't smell right, but it's not too terrible either for now...

Reply:
done